pkg/services/control/server/evacuate.go

```diff
@@ -39,3 +49,62 @@ func (s *Server) EvacuateShard(_ context.Context, req *control.EvacuateShardRequ
 	}
 	return resp, nil
 }
+
+func (s *Server) replicate(addr oid.Address, obj *objectSDK.Object) error {
+	nm, err := s.netMapSrc.GetNetMap(0)
+	if err != nil {
+		return err
+	}
+
+	cid, ok := obj.ContainerID()
+	if !ok {
+		// Container ID is expected to be here. If for some reason it is not,
+		// we log error and return nil to prevent situations
+		// where a shard can't be evacuated because of a single bad/corrupted object.
+		return nil
+	}
+
+	c, err := s.cnrSrc.Get(cid)
+	if err != nil {
+		return err
+	}
+
+	binCnr := make([]byte, sha256.Size)
+	cid.Encode(binCnr)
+
+	ns, err := nm.ContainerNodes(c.Value.PlacementPolicy(), binCnr)
+	if err != nil {
+		return fmt.Errorf("can't build a list of container nodes")
+	}
+
+	nodes := placement.FlattenNodes(ns)
+	bs := (*keys.PublicKey)(&s.key.PublicKey).Bytes()
+	for i := 0; i < len(nodes); i++ {
+		if bytes.Equal(nodes[i].PublicKey(), bs) {
+			copy(nodes[i:], nodes[i+1:])
+			nodes = nodes[:len(nodes)-1]
+		}
+	}
+
+	var res replicatorResult
+	task := new(replicator.Task).
+		WithObject(obj).
+		WithObjectAddress(addr).
+		WithCopiesNumber(1).
+		WithNodes(nodes)
+	s.replicator.HandleTask(context.TODO(), task, &res)
```

Comment:
what if other nodes already have that object? i think that is quite a common case, seems like it is a useless operation then

Reply:
No, we try the fullfill the policy -- we store an object, but will drop it. So we replicate manually here.

Reply:
> we try the fullfill the policy

are you sure? in the replicator we [check](https://github.com/nspcc-dev/neofs-node/blob/c7f85994e550d4c990c08d1ea6529ec99c954770/pkg/services/policer/check.go#L127) if a node contain the object and [select](https://github.com/nspcc-dev/neofs-node/blob/c7f85994e550d4c990c08d1ea6529ec99c954770/pkg/services/policer/check.go#L147) only the nodes that do not have an object before the `HandleTask` call

seems that you take *all* the nodes from the placement, exclude the current one, and try to put the object to the first one *one* time, that is useless if `REP > 1`

i may be wrong

Reply:
This is tricky, in this situation node is expected to be in a maintenance mode, so most likely policer will do the job itself.
Here we only try to _ensure_ that an object is _not lost_. Maintaining a proper amount of copies is a policer task.